skfda/exploratory/visualization/representation.py

```diff
@@ -0,0 +1,320 @@
+
+import matplotlib.cm
+import matplotlib.patches
+
+import numpy as np
+
+from ..._utils import _list_of_arrays, constants
+from ._utils import (_get_figure_and_axes, _set_figure_layout_for_fdata,
+                     _set_labels)
+
+
+def _get_label_colors(n_labels, label_colors=None):
+    """Get the colors of each label"""
+
+    if label_colors is not None:
+        if len(label_colors) != n_labels:
+            raise ValueError("There must be a color in label_colors "
+                             "for each of the labels that appear in "
+                             "sample_labels.")
+    else:
+        colormap = matplotlib.cm.get_cmap()
+        label_colors = colormap(np.arange(n_labels) / (n_labels - 1))
+
+    return label_colors
+
+
+def _get_color_info(fdata, sample_labels, label_names, label_colors, kwargs):
+
+    patches = None
+
+    if sample_labels is not None:
+        # In this case, each curve has a label, and all curves with the same
+        # label should have the same color
+
+        sample_labels = np.asarray(sample_labels)
+
+        n_labels = np.max(sample_labels) + 1
+
+        if np.any((sample_labels < 0) | (sample_labels >= n_labels)) or \
+                not np.all(np.isin(range(n_labels), sample_labels)):
+            raise ValueError("Sample_labels must contain at least an "
+                             "occurence of numbers between 0 and number "
+                             "of distint sample labels.")
+
+        label_colors = _get_label_colors(n_labels, label_colors)
+        sample_colors = np.asarray(label_colors)[sample_labels]
+
+        if label_names is not None:
+            if len(label_names) != n_labels:
+                raise ValueError("There must be a name in  label_names "
+                                 "for each of the labels that appear in "
+                                 "sample_labels.")
+
+            patches = [matplotlib.patches.Patch(color=c, label=l)
+                       for c, l in zip(label_colors, label_names)]
+
+    else:
+        # In this case, each curve has a different color unless specified
+        # otherwise
+
+        if 'color' in kwargs:
+            sample_colors = fdata.n_samples * [kwargs.get("color")]
+            kwargs.pop('color')
+
+        elif 'c' in kwargs:
+            sample_colors = fdata.n_samples * [kwargs.get("color")]
+            kwargs.pop('c')
+
+        else:
+            sample_colors = None
+
+    return sample_colors, patches
+
+
+def plot_hypersurfaces(fdata, chart=None, *, derivative=0, fig=None, axes=None,
```

Comment:
> In my opinion the arguments `fig` and `axes` are redundant and make the api more confusing. Perhaps you should consider in another PR whether to keep only the `chart` parameter.

The `chart` argument is not intended to be used as a keyword argument. The idea is that you can do

```python
fdata.plot(fig)
fdata.plot(axes)
fdata.plot(fig=fig)
fdata.plot(axes=axes)
```
In fact, I think that a better name for `chart` is `canvas`. When Python 3.8 is more used, I would want to change the `chart` parameter to a positional only one.

BTW, the reason for allowing passing the axes is to give the user more control.

> About the name `plot_hypersurfaces`, what do you think about other name such as `plot_graph` or something similar, since it is plotted the [graph of the functions](https://en.wikipedia.org/wiki/Graph_of_a_function).

I agree to this.

Reply:
Renamed function in 0c800cacc0d587b691716c0278c0910f6ce916f2.

Reply:
I thought about the name `canvas`, but as one of the matplotlib objects is called in the same way I thought it might give rise to confusion, but it was a name that I liked more.